File: java-concert/src/main/java/org/example/javaconcert/concert/business/ConcertService.java
package org.example.javaconcert.concert.business;

import java.util.List;

import org.example.javaconcert.concert.domain.ConcertRepository;
import org.example.javaconcert.concert.presentation.request.ConcertRequest;
import org.example.javaconcert.concert.presentation.response.AllConcertResponse;
import org.example.javaconcert.concert.presentation.response.ConcertResponse;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class ConcertService {

	private final ConcertRepository concertRepository;

	public AllConcertResponse findAll() {
		List<ConcertResponse> concertResponses = concertRepository.findAll().stream()
			.map(ConcertResponse::toResponse)
			.toList();

		return new AllConcertResponse(concertResponses);
	}

	@Transactional
	public Long save(ConcertRequest concertRequest) {
		return concertRepository.save(concertRequest.toDomain());
	}
}

File: concert/src/main/kotlin/prac/concert/concert/business/ConcertService.kt
package prac.concert.concert.business

import jakarta.transaction.Transactional
import org.springframework.stereotype.Service
import prac.concert.concert.domain.ConcertRepository
import prac.concert.concert.presentation.request.ConcertRequest
import prac.concert.concert.presentation.response.AllConcertResponse
import prac.concert.concert.presentation.response.ConcertResponse

@Service
class ConcertService(
        private val concertRepository: ConcertRepository
) {

    fun findAll(): AllConcertResponse {
        val concertResponses = concertRepository.findAll()
                .map { concert -> ConcertResponse.toResponse(concert) }

        return AllConcertResponse(concertResponses)
    }

    fun findById(id: Long): ConcertResponse {
        return ConcertResponse.toResponse(concertRepository.findById(id))
    }

    @Transactional
    fun save(concertRequest: ConcertRequest): Long {
        return concertRepository.save(concertRequest.toDomain())
    }
}
